### PR TITLE
Add toString method to Note

### PR DESCRIPTION
**Pull request by clinedome-work**
Was doing some debugging earlier. Most of the other pieces had nice
toString methods, but I couldn't see inside the Note. Quick fix to make
that easier.

Figure we don't test toString methods, though I could add that easily
enough if you wanted.

### CODE REVIEW

File: money-api/src/main/java/com/comcast/money/api/Note.java [hunk 1]
```diff
@@ -273,4 +273,16 @@ public int hashCode() {
         result = 31 * result + (sticky ? 1 : 0);
         return result;
     }
+
+    @Override
+    public String toString() {
+        StringBuilder builder = new StringBuilder();
+        builder.append("Note(");
+        builder.append("name=").append(name);
+        builder.append(", value=").append(value.toString());
```
Comment (paulcleary) [inline money-api/src/main/java/com/comcast/money/api/Note.java]: can a value be null?

Comment (clinedome-work) [inline money-api/src/main/java/com/comcast/money/api/Note.java]: Good call. Checking it out, will push up necessary changes.

**Comment by paulcleary** [comment]
Can you add a simple unit test; one that does a normal and one that does the null?

**Comment by paulcleary** [comment]
👍
